### PR TITLE
Small clickhouse fixes

### PR DESCRIPTION
**Pull request by jwhitaker-gridcog**
Hey guys,

Came across these small fixes while doing an internal project. I've kept each as separate commits.

1) clickhouse is case-sensitive
   - n.b. this could impact consumers (i needed to change behaviour of the schema normalization test), but I think the current behaviour is incorrect (`create table t (a, B)` => `schema.column_names()==['a', 'b']`, this is false as far as clickhouse is concerned).
2) EMPTY property for create table (ref: https://clickhouse.com/docs/en/sql-reference/statements/create/table#from-select-query)
3) convert(date())
4) a bit thornier and without a full fix: parsing `values()` is problematic, I'm not sure of a good way to get this working, so I've done a partial job and added an expectedFailure test to at least document the problem. I'll probably raise an issue for this assuming you guys don't immediately propose a fix.

Thanks for maintaining this!
Jarrad

### CODE REVIEW

File: tests/dialects/test_clickhouse.py [hunk 1]
```diff
@@ -471,6 +474,11 @@ def test_clickhouse(self):
             parse_one("Tuple(select Int64)", into=exp.DataType, read="clickhouse"), exp.DataType
         )
 
+    @unittest.expectedFailure
+    def test_clickhouse_values(self):
+        """clickhouse doesn't support VALUES as a statement; it is, and should be parsed as, a function."""
+        self.validate_identity("SELECT * FROM values('col String', ('why clickhouse why'))")
```
Comment (georgesittas) [inline tests/dialects/test_clickhouse.py:480]: TIL...

**Review by georgesittas — COMMENTED** [review]
Thanks for the PR @jwhitaker-gridcog 👍

What's the issue with `values(...)` right now?

File: sqlglot/dialects/clickhouse.py [hunk 2]
```diff
@@ -106,6 +107,7 @@ def _delta_sql(self: Generator, expression: DATEΤΙΜΕ_DELTA) -> str:
 
 class ClickHouse(Dialect):
     NORMALIZE_FUNCTIONS: bool | str = False
+    NORMALIZATION_STRATEGY = NormalizationStrategy.CASE_SENSITIVE
```
[the remark this answers is not in the document]
Comment (georgesittas) [reply sqlglot/dialects/clickhouse.py:110]: Nvm, I found this ticket - https://github.com/ClickHouse/ClickHouse/issues/33935#issue-1112165779

>  ### A setting to allow case insensitive column, table and database names.
> [...]
> **We also should admit that case sensitiveness (what ClickHouse is doing) is undoubtedly better.**

File: sqlglot/dialects/clickhouse.py [hunk 3]
```diff
@@ -157,6 +159,7 @@ class Tokenizer(tokens.Tokenizer):
             "PREWHERE": TokenType.PREWHERE,
         }
         KEYWORDS.pop("/*+")
+        KEYWORDS.pop("VALUES")
```
[the remark this answers is not in the document]
Comment (jwhitaker-gridcog) [reply sqlglot/dialects/clickhouse.py]: ah good call, thanks. I've removed this as I don't know if we'll need it yet, and added some tests around INSERT VALUES handling to try and prevent future breakage.

**Comment by jwhitaker-gridcog** [comment]
Thanks! I've pushed fixes, I'll squash them back to the 4 commits before merging if all ok.

> What's the issue with values(...) right now?

The issue now is that
`SELECT * FROM values(('abc', 'def'))` (which is valid clickhouse, values-the-table-function)
round trips to
`SELECT * FROM ( VALUES ('abc', 'def') )` (which is not valid clickhouse, values-the-clause).

**Comment by georgesittas** [comment]
Gotcha, I'll take a stab at fixing values. Let's get the other fixes in first.

**Review by georgesittas — APPROVED** [review]
Thanks for the PR!

**Comment by jwhitaker-gridcog** [comment]
nw! fyi my exposure to these was via hacking on sqlmesh,which doesn't yet support clickhouse, but apparently you guys are working on it.
I reckon maybe the clickhouse dialect hasn't really been given a good workout yet because of its absence in sqlmesh. If so I would expect there to be a bit of a torrent of fixups needed if support is added - clickhouse SQL is bloody weird.

I'll just leave this foreboding intro to their docs, best read in the voice of Dr Nick Riviera:
> ClickHouse supports a declarative query language based on SQL that is identical to the ANSI SQL standard in many cases.

**Comment by georgesittas** [comment]
Indeed, ClickHouse is [fun](https://github.com/ClickHouse/ClickHouse/issues/23194)! 😄 

> ### Describe how identifiers in SELECT queries are resolved

Screenshot 2024-08-13 at 1 08 55 AM

But yeah, we'll be happy to accept more contributions if you're up for it! Feel free to join our [Slack](https://tobikodata.com/slack) if you wanna chat.

**Pull request merged**